src/shared/cloudformation/templateRegistry.ts

```diff
@@ -125,18 +128,101 @@ export class CloudFormationTemplateRegistry {
 }
 
 /**
- * Helper function that returns an map of resource names to CloudFormation.Resource objects.
- * Unlike a CloudFormation.TemplateResources object, all resources in this array are guaranteed to be defined.
- * @param templateDatum TemplateDatum object to extract resources from
+ * Filters an array of TemplateDatum objects to those that are tied to the given filepath and handler function.
+ * @param filepath Handler file's path
+ * @param handler Handler function from aforementioned file
+ * @param unfilteredTemplates Array containing TemplateDatum objects to filter
  */
-export function getResourcesFromTemplateDatum(templateDatum: TemplateDatum): Map<string, CloudFormation.Resource> {
-    const map = new Map<string, CloudFormation.Resource>()
-    for (const resourceKey of Object.keys(templateDatum.template.Resources!)) {
-        const resource = templateDatum.template.Resources![resourceKey]
-        if (resource) {
-            map.set(resourceKey, resource)
+export function getTemplatesAssociatedWithHandler(
+    filepath: string,
+    handler: string,
+    unfilteredTemplates: TemplateDatum[] = CloudFormationTemplateRegistry.getRegistry().registeredTemplates
+): TemplateDatum[] {
+    // find potential matching templates
+    return unfilteredTemplates.filter(templateDatum => {
+        return !!getResourceAssociatedWithHandlerFromTemplateDatum(filepath, handler, templateDatum)
+    })
+}
+
+/**
+ * Returns the first Cloudformation Resource in a TemplateDatum that is tied to the filepath and handler given.
+ * Returns undefined if none are found.
+ * TODO: Is there a reason to return more than the first resource?
```

Comment:
It is plausible that the same lambda function code is set up in a template to be used in more than one function resource with different operating contexts/permissions. 

Rather than make this "first resource" decision here, if this method were to return all matches, the calling code can decide if multiple matches are invalid or not. For example, this PR is interested in knowing a detected runtime. If multiple resources are returned, and they all have the same runtime, you've now broadened the set of 'valid' data.

Reply:
Given multiple resources with varying runtimes, what would you propose would be the correct end-user experience?
* Autoselect a runtime as this PR is doing
* Let the user select a runtime from the list of runtimes in the templates (note that this likely puts us in a similar place to where we are now)
* Let the user select a template and resource from a quick pick (potentially provides more options than the above, is more vague in terms of runtimes, and gives a false sense of security that the debug config will be created with the template's context in mind)
* Something else?

Reply:
Via offline conversation:
* Modify this function to return an array of all matching resources. Elevate the picking logic out of this function and make it specific to the codelenses.
* Codelens logic looks for quorum through template files (potentially happens after selection). If not unanimous, provides a list of potential selections.